portable_concurrency/bits/when_all.h

```diff
@@ -27,6 +27,7 @@ class when_all_state final: public future_state<Sequence>
     operations_remains_(sequence_traits<Sequence>::size(futures_) + 1)
   {}
 
+  // TODO overload with allocator
```

Comment:
If direct use of `new` for workaround in libc++ 3.8 bug makes it difficult to make this function allocator aware you can break support of libc++ <3.9. The only reason to have 3.8 as minimal supported version disapeared after android-ndk upgraded their clang toolchain.

Feel free to remove clang 3.8 build in `.travis.yml` and ifdefs in this function if you need.

Reply:
Actually it occurs to me that this might not be a good candidate for making allocator aware anyway.

The TS doesn't seem to specify an overload where in allocator is passed into `when_all` or `when_any`. And at least by the working draft I found (Didn't want to pay for the ISO) doesn't say anything about how an implementation would resolve *an* allocator from a sequence of multiple futures. So I think I might just focus on .then being allocator aware from the parent.